src/lib/watch.ts

```diff
@@ -0,0 +1,67 @@
+import ts = require('typescript')
+import chokidar = require('chokidar')
+import { writeFile, unlink } from './file-util'
+import { LanguageService } from './language-service'
+import { logEmitted, logRemoved, logError } from './logger'
+
+export function watch (
+  dirs: string[],
+  compilerOptions: ts.CompilerOptions = {}
+): chokidar.FSWatcher {
+  const watcher = chokidar.watch(dirs)
+
+  const service = new LanguageService([], {
+    ...compilerOptions,
+    noEmitOnError: true
+  })
+
+  watcher
+    .on('add', onlyVue(file => {
+      service.registerFile(file)
```

Comment:
Maybe it is problematic if the user remove and re-add the same named file.
Internal file version will be reverted to 0 on re-adding.